app/Commands/Dev/Geb/Repl.hs

```diff
@@ -0,0 +1,323 @@
+{-# LANGUAGE QuasiQuotes #-}
+
+module Commands.Dev.Geb.Repl where
+
+import Commands.Base hiding (command)
+import Commands.Dev.Geb.Repl.Format
+import Commands.Dev.Geb.Repl.Options
+import Control.Exception (throwIO)
+import Control.Monad.State.Strict qualified as State
+import Juvix.Compiler.Backend.Geb qualified as Geb
+import Juvix.Compiler.Backend.Geb.Pretty.Values qualified as GebValue
+import Juvix.Compiler.Backend.Geb.Translation.FromSource.Analysis.TypeChecking.Error
+import Juvix.Data.Error.GenericError qualified as Error
+import Juvix.Extra.Version
+import Juvix.Prelude.Pretty qualified as P
+import System.Console.ANSI qualified as Ansi
+import System.Console.Haskeline
+import System.Console.Repline qualified as Repline
+
+type ReplS = State.StateT ReplState IO
+
+type Repl a = Repline.HaskelineT ReplS a
+
+newtype ReplContext = ReplContext
+  { _replContextEntryPoint :: EntryPoint
+  }
+
+data ReplState = ReplState
+  { _replStateContext :: Maybe ReplContext,
```

Comment:
This can be `Maybe EntryPoint` to remove one layer of indirection.

Reply:
Done.